tests/test_shadow_fork.py

```
@@ -72,36 +73,59 @@ def test_create_fails_if_already_active(self, tmp_path):
+        """
         fork = ShadowFork(repo_root=tmp_path)
 
-        with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, "", ""),  # Check for uncommitted changes
-                (False, "", "fatal"),  # Base branch doesn't exist
-            ]
+        existing_refs = {"main"}
+
+        def fake_git(args, **kwargs):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `fake_git`

(ANN202)